src/Search.f90

```diff
@@ -0,0 +1,286 @@
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+!                          Futility Development Group                          !
+!                             All rights reserved.                             !
+!                                                                              !
+! Futility is a jointly-maintained, open-source project between the University !
+! of Michigan and Oak Ridge National Laboratory.  The copyright and license    !
+! can be found in LICENSE.txt in the head directory of this repository.        !
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+!> @brief A Fortran 2003 module implementing some basic search algorithms.
+!>
+!> @par Module Dependencies
+!>  - @ref IntrType "IntrType": @copybrief IntrType
+!>
+!> @author Mitchell Young
+!>    @date 3/30/2017
+!>
+!++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++!
+MODULE Search
+  USE IntrType
+  USE Strings
+  IMPLICIT NONE
+
+  PRIVATE
+
+  PUBLIC :: upperBound
+  PUBLIC :: lowerBound
+  PUBLIC :: find
+  PUBLIC :: binarySearch
+
+  !> @brief Return the index of the first element of a list that compares
```

Comment:
Yeah, I need to check with doxygen on this. I'm of a mind that it's better to take the inconvenience of separating the documentation from the implementations because it's less likely to end up with implementations with disparate semantics and documentation. In a perfect world, we could manage to keep all of this stuff in sync, but... pobody's nerfect.